10up/ElasticPress
# Orders autosuggest foundations

### Description of the Change
The PR implements the foundations for the orders autosuggest for WooCommerce feature. It includes:

- A class for the Orders sub-feature, this incudes:
  - Creating and saving the search template for the order search API on ElasticPress.io.
  - Retrieving and setting a temporary token for authenticated front-end requests to ElasticPress.io.
  - Loading front-end assets.
- The UI.

The class is not instantiated in the WooCommerce feature, so it can not be currently activated.

![Screenshot 2023-02-01 at 12 39 22 am](https://user-images.githubusercontent.com/4100733/215775769-c829e7e6-133f-4a9c-962a-95f5bf4a30aa.png)

Remaining to be implemented, as per #3277, are:

- A new field and ngram that includes the order number, first name, last name, and email address to support partial word matches and highlighting.
- A UI toggle for enabling Orders Autosuggest.
- Logic to index Orders when feature is active, even if Protected Content is disabled.
- Logic to require index when the feature is activated.
- Tests.

### How to test the Change
1. Connect to an EP.io environment that supports the order search API.
2. Activate the WooCommerce and Protected Content features.
3. Add `( new Orders() )->setup();` to the `setup()` method of the WooCommerce feature class.
4. Perform an index.
5. Visit _WooCommerce > Orders_ and start searching for orders. Suggestions should be offered in a listbox below the search form.
 
### Changelog Entry
- Added: Autosuggest for WooCommerce Orders.

### Credits
@JakePT 

### Checklist:
